src/bot.py

```diff
@@ -273,10 +273,14 @@ def on_maintenance(self):
 
         # Check if we are out of tps or need repairing
         need_repair = self._ui_manager.repair_needed()
-        if self._tps_left < random.randint(3, 5) or need_repair or self._config.char["always_repair"]:
+        if self._tps_left < random.randint(3, 5) or need_repair or self._config.char["always_repair"] or self._config.char["always_refill_keys"] & self._config.routes["run_arcane"]:
```

Comment:
I mean it works. But it is a bit weird to use an bitwise and for bool if condition. I think it is more readable if you use  "and" here.